modules/core/04-channel/keeper/upgrade.go

```diff
@@ -122,7 +123,7 @@ func (k Keeper) ChanUpgradeTry(
 
 		// NOTE: OnChanUpgradeInit will not be executed by the application
 
-		k.WriteUpgradeInitChannel(ctx, portID, channelID, upgrade)
+		channel = k.WriteUpgradeInitChannel(ctx, portID, channelID, upgrade)
```

Comment:
this is required as the upgrade sequence is modified here, without this the `channel` variable defined above becomes stale.